rq/job.py

```diff
@@ -733,13 +738,7 @@ def requeue(self):
         """Requeues job."""
         return self.failed_job_registry.requeue(self)
 
-    def delete(self, pipeline=None, remove_from_queue=True,
-               delete_dependents=False):
-        """Cancels the job and deletes the job hash from Redis. Jobs depending
-        on this job can optionally be deleted as well."""
-
-        connection = pipeline if pipeline is not None else self.connection
-
+    def _remove_job_from_container(self, pipeline=None, remove_from_queue=True):
```

Comment:
I think we can rename this method to something more explicit like `job._remove_from_registries()`

Reply:
@selwin I didn't name it that since it can be also be removed from the queue itself as well.

Reply:
There's the `remove_from_queue` argument so I think it's still ok. I'm open to a other suggestions, but I think `container` is a bit weird since we don't have this terminology in RQ.

Reply:
@selwin Fair enough, I'll change it.